commons/src/main/java/kioke/commons/dto/message/notification/NotificationMessageDto.java

```
@@ -0,0 +1,39 @@
+package kioke.commons.dto.message.notification;
+
+import java.util.List;
+import org.springframework.amqp.core.Message;
+import org.springframework.amqp.core.MessageProperties;
+import org.springframework.amqp.support.converter.Jackson2JsonMessageConverter;
+import org.springframework.amqp.support.converter.MessageConversionException;
+import org.springframework.amqp.support.converter.MessageConverter;
+
+public record NotificationMessageDto<T extends NotificationMessagePayload>(
+    NotificationMessageType type, List<NotificationMessageTarget> targets, T payload) {
+
+  public NotificationMessageDto(
+      NotificationMessageType type, List<NotificationMessageTarget> targets, T payload) {
```

To ensure that the payload follows the class referred by the type, let's throw an exception like IllegalArgumentException here if the payload's class type is invalid.